src/backend/XmlHttpRequest.ts

```diff
@@ -29,25 +29,32 @@ export default class XmlHttpRequest extends file_system.BaseFileSystem implement
   public prefixUrl: string;
   /**
    * Constructs the file system.
-   * @param listingUrl The path to the JSON file index generated by
+   * @param listingUrlOrObj index object or the path to the JSON file index generated by
    *   tools/XHRIndexer.coffee. This can be relative to the current webpage URL
    *   or absolutely specified.
    * @param prefixUrl The url prefix to use for all web-server requests.
    */
-  constructor(listingUrl: string, prefixUrl: string = '') {
+  constructor(listingUrlOrObj: string | Object, prefixUrl: string = '') {
     super();
-    if (listingUrl == null) {
-      listingUrl = 'index.json';
+    if (listingUrlOrObj == null) {
+      listingUrlOrObj = 'index.json';
     }
     // prefix_url must end in a directory separator.
     if (prefixUrl.length > 0 && prefixUrl.charAt(prefixUrl.length - 1) !== '/') {
       prefixUrl = prefixUrl + '/';
     }
     this.prefixUrl = prefixUrl;
-    var listing = this._requestFileSync(listingUrl, 'json');
-    if (listing == null) {
-      throw new Error("Unable to find listing at URL: " + listingUrl);
+
+    let listing:Object = null;
```

Comment:
Nit: You should have a space between the declaration and the type annotation, e.g.

`let listing: Object = null;`